### PR TITLE
Impersonating a customer

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Add the feature of impersonating a customer

#### What problem is this solving?
Now the telemarketing can impersonate a customer

#### How should this be manually tested?

[workspace](https://impersonate--storecomponents.myvtex.com/)

#### Screenshots or example usage

image

### Linked to:
- [#135](https://github.com/vtex-apps/store-components/pull/135)
- [#64](https://github.com/vtex-apps/dreamstore/pull/64)

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/locales/es-AR.json [hunk 1]
```diff
@@ -1,28 +1,5 @@
...
-    "login.invalidPassword": "Su contraseña debe contener: 8 caracteres, números y letras minúsculas y mayúsculas",
-    "login.invalidCode": "Inserte un código de acceso de 6 dígitos válida",
-    "login.invalidMatch": "Confirmación de contraseña es incorrecta",
-    "login.wrongCredentials": "Usuario y/o contraseña incorrecta",
-    "login.userBlocked": "Te perdiste tu contraseña 4 veces. Tu usuario ha sido bloqueado por una hora",
-    "login.createPassword": "Crear una nueva contraseña"
-}
\ No newline at end of file
+  "impersonate-customer.message": "Logado como",
+  "impersonate-customer.button": "Login as",
```
Comment (vitoriaHeliane) [inline react/locales/es-AR.json]: Não deveria estar em português?

File: react/locales/pt-BR.json [hunk 2]
```diff
@@ -1,28 +1,5 @@
...
-    "login.invalidPassword": "Sua senha deve conter: 8 caracteres, números e letras minúsculas e maiúsculas",
-    "login.invalidMatch": "Confirmação de senha está incorreta",
-    "login.invalidCode": "Insira os 6 dígitos do código de acesso",
-    "login.wrongCredentials": "Usuário e/ou senha errada",
-    "login.userBlocked": "Você errou sua senha 4 vezes. Seu usuário foi bloqueado por uma hora",
-    "login.createPassword": "Criar uma nova senha"
-}
\ No newline at end of file
+  "impersonate-customer.message": "Logado como",
+  "impersonate-customer.button": "Login as",
```
Comment (andreldsa) [inline react/locales/pt-BR.json]: Pt-br

File: react/locales/es-AR.json (hunk 1 above)
Comment (salesfelipe) [reply react/locales/es-AR.json]: 👍

File: manifest.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
   "vendor": "vtex",
-  "name": "impersonate",
+  "name": "impersonate-customer",
   "version": "0.0.1",
-  "title": "VTEX Impersonate",
+  "title": "VTEX Impersonate Customer",
   "defaultLocale": "pt-BR",
-  "description": "The VTEX Impersonate app",
+  "description": "The VTEX ImpersonateCustomer app",
```
Comment (brunojdo) [inline manifest.json]: split ImpersonateCustomer

File: react/global.css [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.vtex-impersonate-customer {
```
Comment (brunojdo) [inline react/global.css]: Use tachyons pls!

File: react/utils/index.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+const MAX_RETRIES = 3
```
Comment (brunojdo) [inline react/utils/index.js]: A more meaningful name for utils files. Put functions that aren't related in separated files.

File: manifest.json (hunk 3 above)
Comment (salesfelipe) [reply manifest.json]: 👍

File: react/ImpersonateCustomer.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
+import React, { Component, Fragment } from 'react'
+import { Input, Button, Badge } from 'vtex.styleguide'
+
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply react/ImpersonateCustomer.js]: 👍

File: react/global.css (hunk 4 above)
Comment (salesfelipe) [reply react/global.css]: 👍

File: react/utils/index.js (hunk 5 above)
Comment (salesfelipe) [reply react/utils/index.js]: If i separate those functions in different files, I would have a file with 3 lines and only one function, the translate function.

File: react/ImpersonateCustomer.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
+import React, { Component, Fragment } from 'react'
```
[the remark this answers is not in the document]
Comment (brunojdo) [reply react/ImpersonateCustomer.js]: yep, I think we should change others apps to `index.js`

File: react/utils/index.js (hunk 5 above)
Comment (brunojdo) [reply react/utils/index.js]: I don't see a problem with this.

**Pull request merged**